aio_pika/connection.py

```diff
@@ -96,19 +101,37 @@ async def main():
         """
         self.close_callbacks.add(callback, weak=weak)
 
-    def _on_connection_close(self, connection, closing, *args, **kwargs):
-        exc = closing.exception()
-        self.close_callbacks(exc)
+    def _on_connection_close(
+        self, connection: AbstractConnection, closing: asyncio.Future,
+    ) -> None:
         log.debug("Closing AMQP connection %r", connection)
+        exc: Optional[BaseException] = closing.exception()
+        self.close_callbacks(exc)
 
-    async def _make_connection(self, **kwargs) -> aiormq.Connection:
-        connection = await aiormq.connect(self.url, **kwargs)
+        if self.closing.done():
+            return
+
+        if exc is not None:
+            self.closing.set_exception(exc)
+            return
+
+        self.closing.set_result(closing.result())
+
+    async def _make_connection(
+        self, *, timeout: TimeoutType = None, **kwargs: Any
+    ) -> aiormq.abc.AbstractConnection:
+        connection: aiormq.abc.AbstractConnection = await asyncio.wait_for(
+            aiormq.connect(self.url, **kwargs), timeout=timeout,
```

Comment:
aiormq is low level communication driver and the `asyncio.wait_for` should works fine in any cases

Reply:
Let me explain what I mean. The `timeout` parameter in the RPC-related methods is necessary since it uses a slightly more complex cancellation mechanism than sending `CancelledError` to the coroutine.

Therefore, this parameter is present there. In all other cases, async io.wait_for will work.

Reply:
Okay.